code/weapon/weapons.cpp

```diff
@@ -4093,7 +4093,7 @@ void find_homing_object(object *weapon_objp, int num)
 				continue; 
 
 			homing_object_team = obj_team(objp);
-			bool can_lock_on_ship = objp->type != OBJ_SHIP || weapon_can_lock_on_ship(wip, objp->instance);
+			bool can_lock_on_ship = objp->type != OBJ_SHIP || weapon_allowed_lock_restriction_object(wip, objp);
```

Comment:
Since `OBJ_SHIP` is handled in the function, the check here is not needed anymore.